compiler-core/src/warning.rs

```diff
@@ -355,7 +366,29 @@ To match on all possible lists, use the `_` catch-all pattern instead.",
                     extra_labels: vec![],
                 }),
             },
-
+            Warning::DeprecatedSyntax {
+                path,
+                src,
+                warning: DeprecatedSyntaxWarning::DeprecatedEmptyGuardClause { location },
+            } => Diagnostic {
+                title: "Deprecated empty guard clause syntax".into(),
+                text: wrap(
+                    "Empty clause within a guard is deprecated.
+To keep its behaviour, either remove the guard or fill the clause with `True` instead.",
+                ),
+                hint: None,
+                level: diagnostic::Level::Warning,
+                location: Some(Location {
+                    label: diagnostic::Label {
+                        text: Some("This should be followed by a guard or removed!!
```

Comment:
Since the deprecated syntax warnings just suggest fixing the syntax in such a way that behaviour stays the same, I think it makes sense to just suggest removing the `if` here.

```suggestion
                        text: Some("This can be removed"),
```